Assets/Scripts/UI/Animation/SlideAnimation.cs

```diff
@@ -11,37 +11,59 @@
 
 namespace ProjectPorcupine.UI.Animation
 {
-    public class SlideAnimation : MonoBehaviour
+    public class SlideAnimation : MenuAnimation
     {
+        [SerializeField]
+        [Tooltip("Slide in direction.")]
+        private SlideDirection direction;
+
         private PositionTween slideIn;
         private PositionTween slideOut;
-        private Vector2 outPosition;
-        private Vector2 inPosition;
 
-        public void Show()
+        public override void Show()
         {
-            gameObject.SetActive(true);
+            base.Show();
             slideOut.Stop();
             slideIn.Start();
         }
 
-        public void Hide()
+        public override void Hide()
         {
             slideIn.Stop();
             slideOut.Start();
         }
 
-        private void Awake()
+        private void Start()
         {
-            RectTransform rect = GetComponent<RectTransform>();
-            ///Calculates only left to right position.
-            ///TODO: Add All Slide modes(LeftToRight, RightToLeft etc).
-            outPosition = new Vector2(rect.anchoredPosition.x - rect.sizeDelta.x, rect.anchoredPosition.y);
-            inPosition = rect.anchoredPosition;
+            RectTransform rect = gameObject.GetComponent<RectTransform>();
+            Vector2 outPosition = OutPosition(rect, direction);
+            Vector2 inPosition = rect.anchoredPosition;
             rect.anchoredPosition = outPosition;
 
             slideIn = new PositionTween(rect, inPosition);
             slideOut = new PositionTween(rect, outPosition, () => gameObject.SetActive(false));
         }
+
+        private Vector2 OutPosition(RectTransform rect, SlideDirection direction)
+        {
+            float x = rect.anchoredPosition.x;
+            float y = rect.anchoredPosition.y;
+            float width = rect.sizeDelta.x;
+            float height = rect.sizeDelta.y;
+
+            switch (direction)
+            {
+                case SlideDirection.Right:
+                    return new Vector2(x - width, y);
+                case SlideDirection.Left:
+                    return new Vector2(x + width, y);
+                case SlideDirection.Down:
+                    return new Vector2(x, y - height);
+                case SlideDirection.Up:
+                    return new Vector2(x, y + height);
+                default:
+                    throw new System.Exception("Direction not found.");
+            }
```

Comment:
Additionally, since it's an enum, "Unimplemented direction" might be more accurate, since hitting default would require someone adding a new SlideDirection, and not implementing it.

Reply:
That is not necessarily true, there are many ways to get an invalid enum value. The simplest is to just cast an int to the enum.